src/ja/worker/worker_client.py

```diff
@@ -0,0 +1,64 @@
+"""
+Contains classes needed by each WC for providing a DockerClient as well
+as notifying the CS
+"""
+import docker
+
+from ja.common.response_message import RespondMessage
+
+
+class Worker:
+    """
+    The main class on the Worker Client
+    """
+
+    def __init__(self) -> None:
+        """!
+        Initializes the Worker, here the SSHConnection with the Server should
+        take place, initializing the DockerClient as well as starting the event loop in DockerEvents
+         that is going to be notifying the Server when necessary.
+        """
+
+    @property
+    def get_docker_client(self) -> docker.DockerClient:
+        """!
+        @return The DockerClient of the  WC
+        """
+
+
+class WorkerProxy:
+    """
+    This is being called by the SSHConnection
+    """
+
+    def __init__(self) -> None:
+        """!
+        Initialize WorkerProxy
+        """
+
+    def execute_command(self, command: str) -> None:
+        """!
+        This method is called when there is a command to be executed.
+        The command should be converted to an Object.
+        Then the execute method in the commandObject should be called with
+        the DockerClient as a parameter
+        @param command The value of @command
+        """
+
+    def send_respond_message(self, msg: RespondMessage) -> None:
+        """!
+        This method will take a ResponseMessage and will send it to the Central Server.
+        @param msg The @RespondMessage
+        """
+
+
+class DockerEvents:
```

Comment:
Actually I am not sure how to implement this class. I imagined we'd have some event loop and poll an fd or something?

Reply:
I tough we can use https://docker-py.readthedocs.io/en/stable/client.html#docker.client.DockerClient.events
and just filter for example when e container finishes.

Reply:
Didn't know the API looks like that.
Anyway I realized the model depends on what we decide (server asks for updates or wc notifies server)